### PR TITLE
conn: Avoid goroutine leak if the connection closes and there is nothing to write

### PR DESCRIPTION
**Pull request by AlexisMontagne**
Related to https://github.com/gocql/gocql/pull/1194/files/d845a0e2eba7b867634c207bb8f7b84343540fd6#r219705159

### CODE REVIEW

File: conn.go [hunk 1]
```diff
@@ -678,6 +678,11 @@ func (w *writeCoalescer) writeFlusher(interval time.Duration, quit chan struct{}
 		<-timer.C
 	}
 
+	go func() {
+		<-quit
+		w.fcond.Broadcast()
```
Comment (Zariel) [inline conn.go]: Create a method called Close on writeCoalescer like

 ```go
func (w *writeCoalescer) Close() {
    w.fcond.L.Lock()
    defer w.fcond.L.Unlock()
    if !w.running {
        w.fcond.Broadcast()
     }
}
```

Comment (alourie) [reply conn.go]: Pretty sure locking here won't work. Report #1200 shows that there's another goroutine waiting in `writeFlusher` in conn.go#685, which means the w.fcond.L lock has been acquired and will not allow another one. So, if the connection is closed, pretty sure just broadcasting would be enough.

Comment (AlexisMontagne) [reply conn.go]: @Zariel having two closing mechanism (quit channel, and closer implementation) sounds pretty confusing two me. Should we really keep both?

Comment (alourie) [reply conn.go]: @AlexisMontagne pretty sure it's covered now by https://github.com/gocql/gocql/blob/master/conn.go#L708

**Pull request closed without merging**